src/lib/instruction-parser.ts

```diff
@@ -151,13 +152,27 @@ export class InstructionParser {
 
     for (let index = 0; index < swaps.length; index++) {
       const swap: SwapInfo = swaps[index];
-
-      const transferInstructions = this.getInAndOutTransferInstructions(
-        innerInstructions,
-        swap.instructionIndex,
-        swap.isAsk ? swap.inAccount : swap.outAccount,
-        swap.isAsk ? swap.outAccount : swap.inAccount
-      );
+      const swapProgramId =
+        innerInstructions[swaps[index].instructionIndex].programId;
+      const routePlan = routeInfo.routePlan[index];
+      const swapIxName = Object.keys(routePlan.swap)[0];
+      let transferInstructions: any;
+      if (CUSTOM_SWAPS.includes(swapIxName)) {
```

Comment:
this seems like an unnecessary branch. just pass in the `swapProgramId` and let the function decide.

Reply:
You mean I use programId to check if its a custom swap? If I do it with swapProgramId to find if its a custom swap, there will be a problem in few cases. For example stakedex 
supports two swaps

1. Swap::StakeDexStakeWrappedSol (not a custom swap)
2. Swap::StakeDexPrefundWithdrawStakeAndDepositStake (custom swap)

So If I compare just with stakedex address here, first one would also be treated as custom swap which is actually not.